shared-components/src/components/input/BaseInput.tsx

```diff
@@ -0,0 +1,58 @@
+import { DetailedHTMLProps, forwardRef, InputHTMLAttributes } from "react";
+
+import { SubBody } from "@rikstv/shared-components";
+
+export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  className?: string;
+  error?: string | null | false;
+  fullWidth?: boolean;
+  type?:
+    | "button"
+    | "checkbox"
+    | "color"
+    | "date"
+    | "datetime-local"
+    | "datetime"
+    | "email"
+    | "file"
+    | "hidden"
+    | "image"
+    | "month"
+    | "number"
+    | "password"
+    | "radio"
+    | "range"
+    | "reset"
+    | "search"
+    | "submit"
+    | "tel"
+    | "text"
+    | "time"
+    | "url"
+    | "week";
+  autoComplete?: "email" | "new-password" | "off" | "current-password" | "username" | "tel";
+  lightBackground: boolean;
+}
+
+type Props = Omit<DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> & InputProps, "ref">;
+
+export const BaseInput = forwardRef<HTMLInputElement, Props>(
+  ({ className, error, type = "text", fullWidth = false, lightBackground, ...props }, ref) => (
+    <>
+      <input
+        ref={ref}
+        className={`
+        rds-inputfield
+        rds-inputfield-input ${className} ${error ? "inputfield-input--error" : ""} ${
+          fullWidth ? "inputfield-input--full-width" : ""
+        } ${lightBackground ? "rds-light" : ""}
+        `}
+        type={type}
+        {...props}
+      />
+      {error && <SubBody className="inputfield-error-text">{error}</SubBody>}
```

Comment:
Denne må kobles sammen med input feltet med en aria-described-by eller aria-describes eller noe sånt

Reply:
Jeg prøver å ikke være for pedantisk med BEM, men er det litt likevel. Husk hvertfall at alt prefixes med `rds-`